Report: Round values for tables to three decimals

VesnaT opened the pull request:
<!--
Squash the commits, as appropriate. See .github/CONTRIBUTING.md for
more information.

Describe the relevant changes in the commit messages, if they need explaining.
 
If the pull request introduces a new feature or an important bug fix, consider
adding _one_ of the below tags, enclosed in square brackets, in its title:

    ENH, FIX, DOC, WIP

For some example:

    [ENH] CrossValidation: Parallelize computation
    [FIX] NaiveBayesLearner: Don't crash on input containing nan values
    [DOC] Extend documentation for widget development
    [WIP] Working on X ... RFC please

Pull-requests not so tagged will be batched as "other features and bugfixes,"
which is fine for stuff you don't go home and brag to your mother about.
-->

VesnaT committed:
Report: Round values for tables to three decimals

kernc commented on Orange/canvas/report/report.py:
```diff
@@ -216,7 +223,7 @@ def data(role=Qt.DisplayRole,
                         return model.headerData(col if row is None else row,
                                                 orientation, role)
-                    return model.data(model.index(row, col), role)
+                    return fmtnum(model.data(model.index(row, col), role))
```
Would you care to write this as:

``` py
val = model.data(model.index(row, col), role)
return try_(lambda: num_format.format(float(val)), val)
```

?
`try_`, an ingenious invention of mine, does exactly what [this try-except](https://github.com/kernc/orange3/pull/4/files#diff-a8517f05dab824c1e209188aaf6661d5R202) does.

kernc commented on Orange/canvas/report/report.py:
```diff
@@ -196,6 +196,13 @@ def report_table(self, name, table=None, header_rows=0, header_columns=0,
+                return num_format.format(float(s))
+            except:
+                return s
+        fmtnum = fmtnum if num_format else lambda s: s
```
Or here:

``` py
def try_fmtnum(s)
    return try_(lambda: num_format.format(float(s)), s)
```

VesnaT committed:
OWDataTable: Do not override widget's inputs

They should not be overriden, unless there is some (for me) non-obvious reason, in order to make report unittest work for this widget.

kernc commented on Orange/widgets/data/tests/test_owtable.py:
```diff
@@ -0,0 +1,23 @@
+        self.send_signal("Data", self.iris, 1)
+        self.assertEqual(self.widget.tabs.count(), 1)
+        self.send_signal("Data", None, 1)
+        self.assertEqual(self.widget.tabs.count(), 0)
```
Could this be merged into the other method above? :smile:

kernc replied:
Never mind.
